File: step-function-map-in-map/lambda_fns_py/computing_lambda.py
import json
import time
import os
import boto3
import datetime
from botocore.exceptions import ClientError

# # step-fn-activity
client = boto3.client('stepfunctions')
    
def lambda_handler(event, context):
    
    case_Id = event["request"]["caseId"]
    
    tick = 0
    
    start = datetime.datetime.now()
    time.sleep(5)
    end = datetime.datetime.now()
    delta = end - start

    return_obj = event["request"]
    
    return_obj["state"] = "processed in {} seconds".format(delta.total_seconds())

    return return_obj

File: step-function-map-iterator/lambda_fns/computing_lambda.js
'use strict';

// Prepare client to be able to capture SDK 
// https://docs.aws.amazon.com/xray/latest/devguide/xray-sdk-nodejs-awssdkclients.html
const awsxray = require('aws-xray-sdk-core')

// Configure the context missing strategy to do nothing
awsxray.setContextMissingStrategy(() => { });

const aws = awsxray.captureAWS(require('aws-sdk'));
const util = require('util');

const delay = time => new Promise(resolve => setTimeout(resolve, time));

exports.lambda_handler = async (event, context, callback) => {
    // Declare a segment for Xray base on request and case Id to trace
    //{
    //  "request": {
    //    "caseId": "0"
    //  },
    //  "requestId": "123456"
    //}
    const calculationSubsegment = awsxray.getSegment().addNewSubsegment('computing');
    calculationSubsegment.addAnnotation('requestId', event.request.caseId);
    calculationSubsegment.addAnnotation('caseId', event.requestId);

    const start = Date.now();

    // Sleep for 5 seconds to simulate computation
    await delay(5000);

    const end = Date.now();
    const delta = end - start

    // Close the computing calculation segment calculation
    calculationSubsegment.close();

    // Modify original event to add state
    var returnObj = {
        request: event.request,
        requestId: event.requestId,
        state: util.format('processed in %s milliseconds', delta)
    };

    return returnObj;
};